src/standard_wrappers/nft_standards/dip_721.ts

```diff
@@ -32,77 +49,152 @@ export default class ERC721 extends NFT {
     super(canisterId, agent);
 
     this.actor = Actor.createActor(IDL, {
-      agent, canisterId,
+      agent,
+      canisterId,
     });
   }
 
   async getUserTokens(principal: Principal): Promise<NFTDetails[]> {
-    const userTokensResult = await this.actor.getMetadataForUserDip721(principal);
-    const tokens = userTokensResult || [];
+    const userTokensResult = await (async () => {
+      let res;
+
+      try {
+        res = await this.actor.dip721_owner_token_identifiers(principal);
+      } catch (err) {
+        deprecationWarningForDip721LegacyRequests({
+          methodName: 'dip721_owner_token_identifiers',
+        });
+
+        res = await this.actor.ownerTokenMetadata(principal);
```

Comment:
Does this not change in the new version of dip721? If it does, maybe it would be a good idea to move it inside the `try`

Reply:
If a dip721 contract, then either the first would go through, and if not, the second, otherwise is not a dip721 contract.
 
So, that'd be another try/catch, inside a try/catch, as this would be the only case we support as a fallback if the new interface does not exist in the requested canister.